Lib/fontTools/ttLib/tables/_n_a_m_e.py

```diff
@@ -120,6 +120,44 @@ def getDebugName(self, nameID):
 		else:
 			return None
 
+	def getFirstDebugName(self, nameIDs):
+		for nameID in nameIDs:
+			name = self.getDebugName(nameID)
+			if name is not None:
+				return name
+		return None
+
+	def getBestFamilyName(self):
+		# 21 = WWS Family Name
+		# 16 = Typographic Family Name
+		# 1 = Family Name
+		return self.getFirstDebugName((21, 16, 1))
+
+	def getBestSubFamilyName(self):
+		# 22 = WWS SubFamily Name
+		# 17 = Typographic SubFamily Name
+		# 2 = SubFamily Name
+		return self.getFirstDebugName((22, 17, 2))
+
+	def getBestFullName(self):
+		# 4 = Full Name
+		# 6 = PostScript Name
+		for nameIDs in ((21, 22), (16, 17), (1, 2), (4, ), (6, )):
+			if len(nameIDs) == 2:
```

Comment:
Why did you check if the lenght of nameIDs is 2?

Reply:
Hi @moi15moi , thanks for asking. The reason behind is, because the loop starts with the preferred name table ID pairs (21, 22), (16, 17), (1, 2). And if none of them are present (which might be the case in obfuscated web font) then use only the (4, ), (6, ).

Reply:
Ok, thanks.

I am trying to understand why getBestFullName, getBestFamilyName and getBestSubFamilyName does not always return me the good name.

I did this in my program and it seems to always work: https://github.com/moi15moi/FontCollector/blob/main/fontCollector.py#L304-L321

If you wanna try with an font that the name is not correctly decoded, here is an issue where I give one: https://github.com/fonttools/fonttools/issues/2579